test/end-to-end/normalFlow.spec.ts

```diff
@@ -0,0 +1,687 @@
+import { utils, BytesLike, BigNumber } from "ethers";
+import { expect } from "chai";
+import { TestERC20, Bond, BondFactory } from "../../typechain";
+import {
+  expectTokenDelta,
+  getBondContract,
+  getEventArgumentsFromTransaction,
+  getTargetCollateral,
+  getTargetPayment,
+  previewMintAndMint,
+  previewRedeem,
+  redeemAndCheckTokens,
+} from "../utilities";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { bondFactoryFixture, tokenFixture } from "../shared/fixtures";
+import { BondConfigType, BondWithTokens } from "../interfaces";
+import {
+  NonConvertibleBondConfig,
+  ConvertibleBondConfig,
+  UncollateralizedBondConfig,
+  MaliciousBondConfig,
+  ZERO,
+  ONE,
+} from "../constants";
+
+const { ethers, waffle } = require("hardhat");
+const { loadFixture } = waffle;
+
+// Used throughout tests to use multiple instances of different-decimal tokens
+const DECIMALS_TO_TEST = [6, 8, 18];
+
+describe("e2e: Create -> Mint -> Convert -> Pay -> Withdraw -> Mature -> Redeem", () => {
```

Comment:
Is it possible to run this against testnet as well? Similar to `bondflow`

Reply:
Yes - change the forking config or set it in hardhat.config.js - confirmed passing on the forked rinkeby with newly deployed contracts. we could `attach` the factory to an already-deployed address as well, but this flow isn't "the most efficient way" to be deploying bonds. I'd assume if I tried using the already deployed factory in the bondflow file it would fail since the bond arguments probably changed.